rust-wasm-id-allocator/wasm-id-allocator/src/lib.rs

```diff
@@ -205,3 +201,225 @@ impl InteropIdRange {
+            ))
+        );
+        assert!(compressor
+            .normalize_to_session_space(0 as f64, 7.0)
+            .is_nan());
+    }
+
+    #[test]
+    #[should_panic]
+    fn decompress_invalid() {
+        let mut compressor = IdCompressor::new(String::from(SESSION_ID_1)).unwrap();
+        _ = compressor.decompress(1.0);
+    }
+
+    #[test]
+    fn decompress() {
+        let mut compressor = IdCompressor::new(String::from(SESSION_ID_1)).unwrap();
+
+        for _ in 0..5 {
+            compressor.generate_next_id();
+        }
+        let interop_id_range = compressor.take_next_range();
+        _ = compressor.finalize_range(
+            interop_id_range.token,
+            interop_id_range.local,
+            interop_id_range.count,
+        );
+        assert!(compressor.decompress(0.0).is_ok());
+        assert!(compressor.decompress(4.0).is_ok());
+    }
+    #[test]
+    #[should_panic]
+    fn recompress_invalid_uuid_string() {
+        let mut compressor = IdCompressor::new(String::from(SESSION_ID_1)).unwrap();
+        _ = compressor.recompress(String::from("invalid_uuid"));
+    }
+
+    #[test]
+    #[should_panic]
+    fn recompress_unknown_uuid() {
+        let mut compressor = IdCompressor::new(String::from(SESSION_ID_1)).unwrap();
+        _ = compressor.recompress(String::from("0002c79e-b536-4776-b000-000266c252d5"));
+    }
+
+    #[test]
+    fn recompress() {
+        let mut compressor = IdCompressor::new(String::from(SESSION_ID_1)).unwrap();
+
+        for _ in 0..5 {
+            compressor.generate_next_id();
+        }
+        let interop_id_range = compressor.take_next_range();
+        _ = compressor.finalize_range(
+            interop_id_range.token,
+            interop_id_range.local,
+            interop_id_range.count,
+        );
+        let session_id_2 = (StableId::from(SessionId::from_uuid_string(SESSION_ID_1).unwrap()) + 1)
+            .to_uuid_string();
+        assert!(compressor.recompress(session_id_2).is_ok());
+    }
+    #[test]
+    #[should_panic]
+    fn deserialize_invalid() {
+        let bytes: &[u8] = &[1, 2, 1, 0, 1];
+        _ = IdCompressor::deserialize(bytes, String::from(SESSION_ID_1));
+    }
+
+    #[test]
+    fn serialize_deserialize() {
+        let mut compressor = IdCompressor::new(String::from(SESSION_ID_1)).unwrap();
+
+        for _ in 0..5 {
+            compressor.generate_next_id();
+        }
+        let interop_id_range = compressor.take_next_range();
+        _ = compressor.finalize_range(
+            interop_id_range.token,
+            interop_id_range.local,
+            interop_id_range.count,
+        );
+        compressor.generate_next_id();
+        let serialized_local = compressor.serialize(true);
+        assert!(IdCompressor::deserialize(&serialized_local, String::from(SESSION_ID_1)).is_ok());
+        let serialized_final = compressor.serialize(false);
+        assert!(IdCompressor::deserialize(
```

Comment:
probably want to capture the deserialized compressor and compare it to the original using equality